MANUAL.adoc

```diff
@@ -3179,83 +3179,1155 @@ public class LoopExample {
+                        b1.inc(count);
+                    }, b1 -> b1.yield(b1.lt(count, 5)));
+                    b0.return_();
+                });
+            });
+        });
+    }
+}
+----
+//TEST:RUN Continue
+//= Count is 0!
+//= ...divisible by 3!
+//= Count is 1!
+//= Count is 2!
+//= Count is 3!
+//= ...divisible by 3!
+//= Count is 4!
+//TEST:END
+
+[id=try]
+=== `try_`
+
+The `try_(...)` operation encapsulates a block of code which may throw an exception
+and allows for the equivalent of `catch` and `finally` clauses to be defined.
+
+The `try_(...)` method itself accepts a `TryCreator`, which in turn has methods to define
+`catch_(...)` cases and a `finally_(...)` case.
+These work similarly to their Java language equivalents.
+
+For `catch_(...)` clauses, you may specific either a single throwable type or a set of throwable types.
+If a set of throwable types is given as `ClassDesc` values, then the `ClassDesc` of a common
+supertype of all the given types must also be given.
+Otherwise, the type is automatically detected.
+A local variable containing the caught exception will be given to the catch block builder as its second argument.
+The name of the variable is provided to the `catch_(...) method as an argument.
+
+The `finally_(...)` clause, if given, is run at every possible exit path from the body of the `try_(...)`.
+
+//TEST:BEGIN
+.Example of `try_`
+[source,java]
+----
+import java.nio.file.Path;
+
+import io.quarkus.gizmo2.ClassOutput;
+import io.quarkus.gizmo2.Gizmo;
+import io.quarkus.gizmo2.LocalVar;
+import io.quarkus.gizmo2.ParamVar;
+import io.quarkus.gizmo2.desc.MethodDesc;
+
+public class TryExample {
+    public static void main(String[] ignored) {
+        Gizmo gizmo = Gizmo.create(ClassOutput.fileWriter(Path.of("output")));
+        gizmo.class_("Try", cc -> {
+            cc.staticMethod("main", mc -> {
+                ParamVar args = mc.parameter("args", String[].class);
+                mc.body(b0 -> {
+                    LocalVar word = b0.localVar("word", args.elem(0));
+                    b0.try_(t1 -> {
+                        t1.body(b2 -> {
+                            b2.printf("The integer value is %d%n", b2.invokeStatic(
+                                MethodDesc.of(Integer.class, "parseInt", int.class, String.class),
+                                word)
+                            );
+                        });
+                        t1.catch_(NumberFormatException.class, "e", (b2, e) -> {
+                            b2.printf("Value is not a number!%n");
+                        });
+                        t1.finally_(b2 -> {
+                            b2.printf("Thanks for playing!%n");
+                        });
+                    });
+                    b0.return_();
+                });
+            });
+        });
+    }
+}
+----
+//TEST:RUN Try 1234
+//= The integer value is 1234
+//= Thanks for playing!
+//TEST:END
```

Comment:
With this `//TEST:END`, the second assertion is skipped. `//TEST:END` corresponds to `//TEST:BEGIN`, you can have multiple `//TEST:RUN` in it, a single `//TEST:RUN` is finished by anything that doesn't begin with `//=` or `//~`.